chassis/core/src/main/java/com/griddynamics/jagger/util/generators/TerminationGenerator.java

```diff
@@ -0,0 +1,43 @@
+package com.griddynamics.jagger.util.generators;
+
+import com.griddynamics.jagger.engine.e1.scenario.InfiniteTerminationStrategyConfiguration;
+import com.griddynamics.jagger.engine.e1.scenario.IterationsOrDurationStrategyConfiguration;
+import com.griddynamics.jagger.engine.e1.scenario.TerminateStrategyConfiguration;
+import com.griddynamics.jagger.user.test.configurations.termination.JTermination;
+import com.griddynamics.jagger.user.test.configurations.termination.JTerminationBackground;
+import com.griddynamics.jagger.user.test.configurations.termination.JTerminationDuration;
+import com.griddynamics.jagger.user.test.configurations.termination.JTerminationIterations;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * @author asokol
+ *         created 11/6/16
+ *         Generates {@link TerminateStrategyConfiguration} entity from user-defined {@link JTermination} entity.
+ */
+class TerminationGenerator {
+
+    static TerminateStrategyConfiguration generateTermination(JTermination jTermination) {
```

Comment:
Instead of using such heavy and hardly-read method, I recommend to split it into 3 methods: 

- generateTerminationIterations(...)
- generateTerminationDuration(...)
- generateTerminationBackground(...)

And try to avoid cast operations, really hard to read(
You could create a temp variable with cast and use it further.